docs/madr/decisions/008-mesh-logging.md

```diff
@@ -0,0 +1,479 @@
+
+In the future,
+when `spec.targetRef` is `Mesh|MeshSubset|MeshService|MeshServiceSubset` we can have `MeshHTTPRoute` as an Outbound route,
+and it can only have `to` (`from` always goes from the application).
+
+### Examples
+
+MeshLogging can be an `inbound`, or `outbound` or `inbound/outbound` policy
+meaning it can have `from`, `to`, or both `from` and `to` sections.
+
+#### Only inbound
+
+```yaml
+type: MeshLogging
+mesh: default
+spec:
+  targetRef:
+    kind: MeshService
+    name: web-backend
+  from:
+    - targetRef:
+        kind: MeshService
+        name: web-frontend
+```
+
+This will log all the inbound traffic from `web-frontend` in `web-backend`.
+
+#### Only outbound
+
+```yaml
+type: MeshLogging
+mesh: default
+spec:
+  targetRef:
+    kind: MeshService
+    name: web-backend
+  to:
+    - targetRef:
+        kind: MeshService
+        name: web-queue
+```
+
+This will log all the outbound traffic to `web-queue` in `web-backend`.
+
+#### Both inbound and outbound
+
+```yaml
+type: MeshLogging
+mesh: default
+spec:
+  targetRef:
+    kind: MeshService
+    name: web-backend
+  from:
+    - targetRef:
+        kind: MeshService
+        name: web-frontend
+  to:
+    - targetRef:
+        kind: MeshService
+        name: web-queue
+```
+
+This will log all the inbound traffic from `web-frontend` and outbound traffic to `web-queue` in `web-backend`.
+
+### Backends
+
+#### Keep backends in the `Mesh` resource for now, move them into `MeshLoggingBackend` policy in the future
+
+For now new Policy `MeshLogging` will reuse the backend definitions from `Mesh`.
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: Mesh
+metadata:
+  name: default
+spec:
+  logging:
+    # MeshLogging policies may leave the `backend` field undefined.
+    # In that case the logs will be forwarded into the `defaultBackend` of that Mesh.
+    defaultBackend: file
+    # List of logging backends that can be referred to by name
+    # from MeshLogging policies of that Mesh.
+    backends:
+      - name: logstash
+        type: tcp
+        conf:
+          address: 127.0.0.1:5000
+      - name: file
+        type: file
+        conf:
+          path: /tmp/access.log
+---
+type: MeshLogging
+mesh: default
+spec:
+  targetRef:
+    kind: MeshService
+    name: web-backend
+  from:
+    - targetRef:
+        kind: MeshService
+        name: web-frontend
+      default:
+        backends:
+          - name: logstash
+            type: reference # this is the default and only supported value atm, people can ignore this
+```
+
+In the future we will introduce `MeshLoggingBackend` policy that would hold that data.
+Definitions there will have precedence over definitions in `Mesh`.
+In the future the user will re-create (or maybe we could do this automatically) equivalent `MeshLoggingBackend` policies.
+
+##### Positive Consequences 
+
+* easier to implement
+
+##### Negative Consequences
+
+* possibly more work for the user in the future
```

Comment:
`* it's impossible to create an observability role using RBAC. You need to give access to the whole mesh to such a person`